### DISCUSSION TITLE
No entrypoint error though the .dll file is getting created

### DISCUSSION THREAD

**Discussion by AshwinSridharan0410** [paketo-buildpacks/feedback#266] [.NET Core Team]
I am trying to build a dotnet azure functions app of .net6.

It is throwing me an error like this :-
**no entrypoint [abc-Function.dll] found: stat /workspace/abc-Function.dll: no such file or directory**

This is my .csproj file :-
![image](https://github.com/paketo-buildpacks/feedback/assets/113506379/803a7aaf-ad45-43c5-9d73-58cd464a42f1)

The command that I used:- (I am building it using the dotnet-core buildpack)

 **pack build azure-functions --buildpack dotnet-core --path=.**
![image](https://github.com/paketo-buildpacks/feedback/assets/113506379/227dade7-1b32-4bc9-93cb-37ef0509a4a2)
![image](https://github.com/paketo-buildpacks/feedback/assets/113506379/31025a00-0ccd-4c8b-b925-f12b0c1fe40f)
![image](https://github.com/paketo-buildpacks/feedback/assets/113506379/2acad214-8bbd-405b-9d8a-6d82696c4b15)

Would like to know your views here  @sophiewigmore  @andymoe  @nebhale @anthonydahanne @ryanmoran @eddumelendez

**Comment by sophiewigmore** [comment]
@AshwinSridharan0410 what's the structure of your app? We recommend using `BP_DOTNET_PROJECT_PATH` to specify the path to the startup project you want to build. See https://paketo.io/docs/howto/dotnet-core/#using-bp_dotnet_project_path

Is your app a multi-project solution? If so, you should definitely use the `BP_DOTNET_PROJECT_PATH` env. var. [More documented here](https://paketo.io/docs/howto/dotnet-core/#build-one-project-in-a-multi-project-solution)

**Comment by AshwinSridharan0410** [comment]
Thanks @sophiewigmore . It worked . 
Logs of that build 
![image](https://github.com/paketo-buildpacks/feedback/assets/113506379/e12fd468-fcf5-4b81-9669-355a7c65bf80)
![image](https://github.com/paketo-buildpacks/feedback/assets/113506379/e9b023ee-1ded-4084-8551-e215dbee8b85)
![image](https://github.com/paketo-buildpacks/feedback/assets/113506379/5fd6439a-2c38-431c-a976-7ae314f53491)
But when I run the image with this command 
docker run -p 6410:80 -p 6411:443 --env "ASPNETCORE_ENVIRONMENT=Development" --env "DOTNET_ROOT=/layers/paketo-buildpacks_dotnet-core-aspnet-runtime/dotnet-core-aspnet-runtime" --env "PATH=$PATH:$DOTNET_ROOT" azurefunctions002
or
docker run -it azurefunctions002 /bin/sh

I get this error:-
Setting ASPNETCORE_URLS=http://0.0.0.0:8080
ERROR: failed to launch: path lookup: exec: "/workspace/abc-Function": permission denied

**Reply by sophiewigmore** [reply]
what happens if you run `docker run -it --entrypoint launcher  azurefunctions002  ls -al /workspace`

**Reply by sophiewigmore** [reply]
what does your builder look like?

**Reply by sophiewigmore** [reply]
What are the permissions on you app directory?
